### ISSUE TITLE
Provide option to get list of TextFormField that failed validation 

### ISSUE THREAD

**Issue by edwin-alvarez** [flutter/flutter#55959]
What I'm trying to achieve is to focus the first TextFormField with validation error after _formKey.currentState.validate() is executed. but, if we have the option to get the list of  TextFormField which failed validation that will also allow creating custom widgets to display the full list of error as a summary
```dart
final GlobalKey<FormState> _formKey = new GlobalKey<FormState>();

  void _validateInputs() {
    final form = _formKey.currentState;
    if (form.validate()) {
      form.save();
    } else {
///var errors = form.getErrors() <-- will be great
      setState(() {
        FocusScope.of(context).requestFocus(_myFocusNodeName);
///FocusScope.of(context).requestFocus(errors[0].focusNode);
        _autoValidate = true;
      });
    }
  }
```

**Comment by HansMuller** [comment]
CC @justinmc

**Comment by justinmc** [comment]
Currently, I think you'd need to maintain your own list of form field keys and iterate them to find the first that isn't valid.

The idea of a FormState.getErrors would be easy enough to implement.  I wonder why the fields in the form are kept private currently.

Here's the implementation of FormState.validate for reference:

https://github.com/flutter/flutter/blob/b7586d873a6b63a1ea07963e480b1bf8154905f0/packages/flutter/lib/src/widgets/form.dart#L206

**Comment by edwin-alvarez** [comment]
Hi guys. thanks for the fast answer.  This morning I also was looking into that class and noticed the collection of fields was private :(, not sure why neither.

then, I tried to create a child class that extends FormState, but since _fields is private the child class was not able to access it neither. then I decided to copy the  
FormState class to my project. but that class also uses _FormScop. which is also private.

at this point, I abandoned the idea and declared an ugly and dirty solution of having a middle man (temp FocusNode) which is set when the Textformfield validator detects an error.

if you guys can make the _fields property public or implement the geterror call, also I noticed the _fields is a collection of FormFieldState, will be great if get error return the list of TextFormField so we can get the focus node
